metadata-ingestion/src/datahub/ingestion/source/usage/usage_common.py

```diff
@@ -42,8 +43,11 @@ class GenericAggregatedDataset(Generic[ResourceType]):
     def add_read_entry(
         self, user_email: str, query: Optional[str], fields: List[str]
     ) -> None:
+
         self.readCount += 1
-        self.userFreq[user_email] += 1
+        if not self.ignore_user_emails or user_email not in self.ignore_user_emails:
+            self.userFreq[user_email] += 1
+
         if query:
```

Comment:
Why are we not dropping the query made by an ignored user? Should we not just drop the entry altogether, if we don't want the data from the user pollute the usage stats?

Reply:
To add in here, when we first discussed this feature we agreed to not drop them from totals as it was a heavier touch operation. I think we have enough signal to know that it would make sense to drop these users from sum total as well.

Reply:
I updated the logic to skip these users from every count.